tests/golden/test_schedules/test_stage_mem_okay.txt

```diff
@@ -0,0 +1,9 @@
+def foo(x: i8[10, 10, 10] @ DRAM):
+    y: i8 @ DRAM
+    x_tmp: i8[10, 10] @ DRAM
+    for i0 in seq(0, 10):
+        for i1 in seq(0, 10):
+            x_tmp[i0, i1] = x[i0, 0, i1]
+    for i in seq(0, 10):
+        x[i, i, i] = 1.0
```

Comment:
This also seems wrong. `x[i, i, i]` overlaps with `x[0:10, 0, 0:10]` when `i = 0`. This particular code example happens be okay, but if you replaced `x[2, 0, 3]` with `x[0, 0, 0]`, then `stage_mem` would break correctness.

Reply:
Ugh, thanks for noticing this!